assets/js/view/buy_trains.rb

```diff
@@ -95,7 +95,7 @@ def from_depot(depot_trains)
 
     def other_trains(other_corp_trains)
       other_corp_trains.flat_map do |other, trains|
-        trains.map do |train|
+        trains.group_by(&:name).each_pair.map do |name, same_trains|
```

Comment:
i don't think you need each pair, and can you rename same_trains to group

Reply:
group_by produces a hash of {name1:trainarray1, name2:trainarray1, ...} which I couldn't find a way to iterate without first turning it into an array with each_pair.
happy to do the rename.

Reply:
Well, apparently I can do `.group_by().map()` despite none of the docs I can find explaining why or how that would work. So... done.